src/entity_metadata.js

```diff
@@ -0,0 +1,59 @@
+import entity_metadata from './entity_metadata.json'
+
+export const villager_types = {
+  DESERT: 0,
+  JUNGLE: 1,
+  PLAINS: 2,
+  SAVANNA: 3,
+  SNOW: 4,
+  SWAMP: 5,
+  TAIGA: 6,
+}
+
+export const villager_professions = {
+  NONE: 0,
+  ARMORE: 1,
+  BUTCHER: 2,
+  CARTOGRAPHER: 3,
+  CLERIC: 4,
+  FARMER: 5,
+  FISHERMAN: 6,
+  FLETCHER: 7,
+  LEATHERWORKER: 8,
+  LIBRARIAN: 9,
+  MASON: 10,
+  NITWIT: 11,
+  SHEPERD: 12,
+  TOOLSMITH: 13,
+  WEAPONSMITH: 14,
+}
+
+function to_bitmask(bitflags, object) {
+  return Object.entries(bitflags).reduce(
+    (result, [name, value]) => result | (object[name] << value),
+    0
+  )
+}
+
+export function to_metadata(entity, metadatas) {
+  const { parent, metadata } = entity_metadata[entity] ?? {}
+  if (!metadata) throw new Error(`${entity} doesn't have any metadata`)
```

Comment:
An entity can have a parent and no metadata for example the silverfish https://wiki.vg/Entity_metadata#Silverfish
```suggestion
  if (!(entity in entity_metadata)) throw new Error(`${entity} doesn't have any metadata`)
  const { parent, metadata } = entity_metadata[entity]
```